pliny/pliny.go

```diff
@@ -0,0 +1,169 @@
+package pliny
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"fmt"
+
+	"net/url"
+
+	"encoding/json"
+	"io"
+
+	"bytes"
+)
+
+type OAuthResponse struct {
+	ExpiresIn   int    `json:"expires_in"`
+	AccessToken string `json:"access_token"`
+}
+
+type RestConnection struct {
+	AccessToken    string
+	APIVersion     string
+	ClientId       string
+	ClientSecret   string
+	Host           string
+	Password       string
+	Username       string
+}
+
+type RestApi struct {
+	Connection *RestConnection
+}
+
+// Logs a given user into a target Skuid Platform site and returns a RestApi connection
+// that can be used to make GET / POST requests
+func Login(host string, username string, password string, apiVersion string, verboseLogs bool) (api *RestApi, err error) {
+
+	if apiVersion == "" {
+		apiVersion = "1"
+	}
+
+	if verboseLogs {
+		fmt.Println("Logging in to Skuid Platform... ")
+		fmt.Println("Host: " + host)
+		fmt.Println("Username: " + username)
+		fmt.Println("Password: " + password)
+		fmt.Println("API Version: " + apiVersion)
+	}
+
+	conn := RestConnection{
+		Host:			host,
+		Username:		username,
+		Password:		password,
+		APIVersion:		apiVersion,
+	}
+
+	err = conn.Refresh()
+
+	if err == nil {
+		api = &RestApi{
+			Connection: &conn,
+		}
+	}
+
+	if verboseLogs {
+		fmt.Println("Login successful! Access Token: " + conn.AccessToken)
+	}
+
+	return api, err
+}
+
+// Used to obtain an OAuth2 access_token
+func (conn *RestConnection) Refresh() (err error) {
```

Comment:
Since you're initializing `err` below in your `resp, err := http.PostForm`, you don't strictly _need_ to define `(err error)` for your return. You could simply do `Refresh() error {`. No change required: but just an FYI.